src/pages/MyData/index.tsx

```diff
@@ -1,31 +1,53 @@
-import React, { useContext } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
-
-import UserContext from 'pages/Login/UserContext';
-import InfoBanner from 'components/UI/InfoBanner';
 import Loading from 'components/UI/Loading';
 import FetchError from 'components/UI/FetchError';
 import EMGTable from 'components/UI/EMGTable';
-import useMGnifyData from 'hooks/data/useMGnifyData';
-import { MGnifyResponseList } from 'hooks/data/useData';
 import { getBiomeIcon } from 'utils/biomes';
 import useQueryParamState from 'hooks/queryParamState/useQueryParamState';
 
-const MyData: React.FC = () => {
-  const { isAuthenticated } = useContext(UserContext);
+import useProtectedApiCall from 'hooks/useProtectedApiCall';
+import { MGnifyResponseList } from 'hooks/data/useData';
 
+const MyData: React.FC = () => {
   const [page] = useQueryParamState('page', 1, Number);
-  const [order] = useQueryParamState('order', '');
-  const [pageSize] = useQueryParamState('page_size', 25, Number);
 
-  const { data, loading, isStale, error, downloadURL } = useMGnifyData(
-    isAuthenticated ? 'mydata' : null,
-    {
-      page,
-      ordering: order,
-      page_size: pageSize,
-    }
-  );
+  const protectedAxios = useProtectedApiCall();
```

Comment:
Cool :) I believe we need to use this in *many* more places, though. With the current login, a Cookie is set on the client and is sent with *every* request to the API. This means that once a user follows a link from their My Data page say, they can browse a Study page and their API requests to that private data will be correctly authenticated. And then following through to assemblies, analyses etc – all of those require authenticated API requests.

Now that we have the token auth, every call to the API needs to include the `Authorization: Bearer ` header instead of the cookie (if I understand right).

So I think we need to modify the `useData` or `useMgnifyData` hooks to use this axios based protectedApiCall, which probably means replacing all the `fetch` with `axios` (or implementing this with `fetch` instead).